### PR TITLE
Yearn deposit yield token

### PR DESCRIPTION
**Pull request by chuddster**

### CODE REVIEW

File: test/RibbonThetaYearnVault.ts [hunk 1]
```diff
@@ -933,92 +933,169 @@ function behavesLikeRibbonOptionsVault(params: {
       });
     });
```
Comment (chuddster) [inline test/RibbonThetaYearnVault.ts:935]: need to finish

File: test/RibbonThetaYearnVault.ts [hunk 2]
```diff
@@ -83,8 +83,8 @@ describe("RibbonThetaYearnVault", () => {
     tokenDecimals: 18,
     isPut: false,
     gasLimits: {
-      depositWorstCase: 140390,
-      depositBestCase: 122555,
+      depositWorstCase: 154539,
+      depositBestCase: 133598,
```
Comment (kenchangh) [inline test/RibbonThetaYearnVault.ts:87]: Question out of scope of this PR, why is the gas use for deposits so much [worse than the vanilla vaults](https://github.com/ribbon-finance/ribbon-v2/blob/master/test/RibbonThetaVault.ts#L79-L80)?

**Review by kenchangh — APPROVED** [review]
LGTM

File: test/RibbonThetaYearnVault.ts (hunk 2 above)
Comment (kargakis) [reply test/RibbonThetaYearnVault.ts:87]: :(

**Pull request merged**